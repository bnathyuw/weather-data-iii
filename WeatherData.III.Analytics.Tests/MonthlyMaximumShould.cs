using FluentAssertions;
using NUnit.Framework;
using WeatherData.III.Objects.Adla;
using static WeatherData.III.Analytics.Tests.AnalyticsTestJig;

namespace WeatherData.III.Analytics.Tests
{
    [TestFixture]
    public class MonthlyMaximumShould
    {
        private const string Location = "narnia";
        private const int Year = 1999;
        private const int Month = 5;
        private const double MaximumTemperature = 12.3;

        [SetUp]
        public void SetUp()
        {
            CreateInputFolder("metOfficeObservations");
        }

        [Test]
        public void ExtractExpectedValues()
        {
            var input = new MaximumTemperatureInput
            {
                Year = Year,
                Month = Month,
                MaximumTemperature = MaximumTemperature
            };
            Write($"input\\metOfficeObservations\\{Location}data.txt", input);

            Run(AnalyticsScript("monthlyMaximum.usql"));

            var output = Read<MaximumTemperatureOutput>("output\\monthlyMaximum.csv");
            output.Should().Be(new MaximumTemperatureOutput
            {
                Location = Location,
                Year = Year,
                Month = Month,
                MaximumTemperature = MaximumTemperature
            });
        }

        [Test]
        public void SelectMaximumValueFromInput()
        {
            Write($"input\\metOfficeObservations\\{Location}data.txt",
                new MaximumTemperatureInput {Year = Year, Month = Month, MaximumTemperature = 1},
                new MaximumTemperatureInput {Year = Year, Month = Month, MaximumTemperature = 2},
                new MaximumTemperatureInput {Year = Year, Month = Month, MaximumTemperature = 3},
                new MaximumTemperatureInput {Year = Year, Month = Month, MaximumTemperature = 4},
                new MaximumTemperatureInput {Year = Year, Month = Month, MaximumTemperature = 5});

            Run(AnalyticsScript("monthlyMaximum.usql"));

            var output = Read<MaximumTemperatureOutput>("output\\monthlyMaximum.csv");

            output.Should().Be(new MaximumTemperatureOutput
            {
                Location = Location,
                Year = Year,
                Month = Month,
                MaximumTemperature = 5
            });
        }
    }
}
