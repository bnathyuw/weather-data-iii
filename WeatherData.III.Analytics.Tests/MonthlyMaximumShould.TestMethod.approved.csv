"sample",2010,1,10.3
"sample",2010,2,13.3
"sample",2010,3,15.6
"sample",2010,4,18.9
"sample",2010,5,20.4
"sample",2010,6,24.6
"sample",2010,7,26.1
